### PR TITLE
Preserve residue IDs, insertion codes, and bond orders in OpenMM topology conversions

### PR DESCRIPTION
**Pull request by jpthompson17**
This PR modifies `parmed.openmm.load_topology()` and `Structure.topology.getter()` in order to preserve additional topology information.

Changes to `parmed.openmm.load_topology()`:
- If an OpenMM residue `id` is the string representation of an integer, set the corresponding ParmEd residue `number` to that integer.
- ParmEd residue `insertion_code` <- OpenMM residue `insertionCode`
- If an OpenMM bond carries `type` data, use it to set the corresponding ParmEd bond `order`. Otherwise, use the OpenMM bond `order` (converted to a float).

Changes to `Structure.topology.getter()`:
- OpenMM residue `id` <- string representation of ParmEd residue `number`
- OpenMM residue `insertionCode` <- ParmEd residue `insertion_code`
- Use ParmEd bond `order` to determine OpenMM bond `type` and `order`.

### CODE REVIEW

File: parmed/structure.py
Comment (swails) [inline parmed/structure.py:1726]: I've also recently added a `qualitative_type` that is basically a translation of the RDKit bond types.  Not sure if that would be useful here or not.

**Review by swails — APPROVED** [review]
Thanks!

**Comment by swails** [comment]
The build failures are fixed in #1315, so once that is merged this can be rebased and should run to completion.

The failures are a result of downstream updates in dependencies (scipy, AmberTools, and OpenMM).

File: parmed/structure.py
Comment (jpthompson17) [reply parmed/structure.py:1726]: Hmm... maybe. Would it make sense to add
```py
qualitative_type = None
if b.type is not None:
    name = repr(b.type).upper()
    if hasattr(QualitativeBondType, name):
        qualitative_type = getattr(QualitativeBondType, name)
```
to `parmed.openmm.load_topology()`, and do something like
```py
type = None
if bond.qualitative_type is not None:
    name = bond.qualitative_type.name.lower().capitalize()
    if hasattr(app, name):
        t = getattr(app, name)
        if t in bond_orders_by_openmm_type:  # ensure t is a recognized type
            type = t
else:  # try determining the type from the bond order
    for t, o in bond_orders_by_openmm_type.items():
        if math.isclose(bond.order, o):
            type = t
```
in `Structure.topology.getter()`?

Comment (swails) [reply parmed/structure.py:1726]: I'm tempted to say let's hold off on that until and unless it proves useful, and let's just get this change merged.

**Pull request merged**
